shared/domains/okTurtles/events/index.js

```diff
@@ -4,26 +4,31 @@
 // Domain: Event publish/subscribe
 // =======================
 
-import sbp from '../../../sbp'
+import sbp from '../../../sbp.js'
 
-const SELECTORS = {
-  // TODO: add ability to unregister listeners
+const listenKey = evt => `events/${evt}/listeners`
+
+export default sbp('sbp/selectors/register', {
   'okTurtles.events/on': function (event: string, handler: Function) {
-    sbp('okTurtles.data/add', `events/${event}/listeners`, handler)
+    sbp('okTurtles.data/add', listenKey(event), handler)
```

Comment:
Cool work here! The new logs on the console can be helpful!

I played a little with `sbp()` locally during this CR and some questions came to my mind:

- Does `sbp` have a naming conventions for the selectors? I saw selectores in camelCase and kebab-case, (`backend/publishLogEntry` vs `gi/contract/create-action`) and I believe we should sticky with one, similar to what Vue did with events names (kebab-case).

- Do we have guidelines to decide the scope of a selector? `backend`, `gi`, `okturtles`, `namespace`, `state`, etc... It's not hard to give the same name twice. How do you plan to avoid this besides _using an SBP logging facility_?

- I registered my first `sbp()` using `'sbp/selectors/register'` and `'okTurtles.events/on'` with ease!

```js
// GroupDashboard.vue

created () {
  sbp('sbp/selectors/register', {
    'test/selector-1': (origin) =>
      console.log('My first selector from ' + origin)
  })

  sbp('okTurtles.events/on', 'hello', (origin) =>
    console.log('My first hello emit from ' + origin)
  )
}
```

```js
// Voting.vue

handleVoteFor () {
  sbp('test/selector-1', 'selector-voting')
  sbp('okTurtles.events/emit', 'hello', 'emit-voting')

  // ...
},
```

And both logged 

```
> My first selector from selector-voting
> My first hello emit from emit-voting
```

So, it looks like they both do the same but with different usages. Can you tell/explain me the core difference between both and when should I use one or another?

Reply:
Yes indeed they look somewhat similar but the semantics are rather different, for example:

- You can emit an event that has no listeners but you cannot call a selector that hasn't been registered
- A single event can have multiple listeners but a selector can have only one function registered for it
- When you emit an event there is no return value, but when you can a selector there is

etc.

Generally speaking, for all Vue related things you should probably use events only. Selectors should be registered only for things that can be conceptualized as services. In this way events are a bit more "light weight" than selectors.

By separating events and selectors we also have the ability to do things like add SBP filters just for events that are emitted, so for example, if we wanted to re-integrate the Vue.js Chrome debugger extension with our events system, so that okTurtles.events appear in the Vue events debugger, we could add an SBP filter to emit Vue events for every okTurtles.events that were emitted, while ensuring that SBP calls do not appear in that list.

> Does sbp have a naming conventions for the selectors? I saw selectores in camelCase and kebab-case, (backend/publishLogEntry vs gi/contract/create-action) and I believe we should sticky with one, similar to what Vue did with events names (kebab-case).

I will think on this. I agree it's important but it's also a decision that requires a lot of thought and I wanted to get this PR in quickly.

> Do we have guidelines to decide the scope of a selector? backend, gi, okturtles, namespace, state, etc... It's not hard to give the same name twice

So that would be called the "domain" of a selectors. And no, at the moment we haven't figured out good guidelines for that, besides making sure to only register domains for clearly defined services, and avoid name collisions.

I am considering prefixing `backend` etc. under the `gi` domain, or `gi.frontend` domain, but this is something that I'll need to give more thought to.

Reply:
Thank you for the extended clarification. It would be cool to add the `emit` vs `register` sbp usage somewhere on our docs :)